Max out smarty grumpiness in unit tests - obviously failed - need to figure out why (tests should have failed)

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
Max out smarty grumpiness in unit tests

Before
----------------------------------------
Happy smarty

After
----------------------------------------
Frowny smarty

Technical Details
----------------------------------------

Comments
----------------------------------------

demeritcowboy commented on CRM/Core/Smarty.php:
```diff
@@ -158,6 +158,17 @@ private function initialize() {
+   * @return bool
+   */
+  private static function isTestEnvironment(): bool {
+    return CRM_Core_Config::singleton()->userFramework === 'CRM_Utils_System_UnitTests';
```
Is this right? Would it just be 'UnitTests'?

demeritcowboy replied:
Also bootstrap [defines CIVICRM_TEST](https://github.com/civicrm/civicrm-core/blob/8104b62cb9b474aae6520139b02dd98d2eb9dc53/tests/phpunit/CiviTest/bootstrap.php#L17) and I've seen that used elsewhere.

eileenmcnaughton committed:
Max out smarty grumpiness

demeritcowboy commented:
Just looking at the log so far this broke jenkins while still at the letter "A", so most of the `CRM_` tests didn't run.

👍

eileenmcnaughton commented:
Looks like it started failing here

not ok 224 - Error: CRM_Case_BAO_CaseTest::testCaseActivity
not ok 225 - Error: CRM_Case_BAO_CaseTest::testAddCaseToContact
not ok 226 - Error: CRM_Case_BAO_CaseTest::testSortByCaseContact

demeritcowboy commented:
I see `not ok 165 - Error: CRM_Activity_Form_ActivityTest::testActivityCreate`

eileenmcnaughton commented:
UG - that one passed for me locally - even running from the start!

There is one known test failing thing open - #22470 is what is blocking tests in https://github.com/civicrm/civicrm-core/pull/22466  - not sure if that could also affect  here

demeritcowboy commented:
Ok I can look at 22470.

eileenmcnaughton commented:
I've found the thing in the activity tests (or at least one thing)

eileenmcnaughton commented:
This looks like the current fail -

![image](https://user-images.githubusercontent.com/336308/149731285-e5720973-9018-4a6e-b46c-679ea773638f.png)


I think #22470 will get past that notice

eileenmcnaughton committed:
Ensure tags fields are assigned to tpl for email form test

eileenmcnaughton commented:
currently falling over on the isset on totalTaxAmount in the contribution template - this value could be cast from the `{contribution.tax_amount}` token - but needs to be assigned in an unformatted style #22547 would be the pre-requisite to fixing

eileenmcnaughton commented:
ok - this would fix the current fail https://github.com/civicrm/civicrm-core/pull/22560

eileenmcnaughton committed:
Temporary fix for enotice on balance amount

Also fix test to only expect balance if it is sane for there to be one...

eileenmcnaughton committed:
Re-fix enotice on templateSelected

The empty check on templateSelected was added to stop tests failing on a smarty notice.

However, this still causes a smarty notice (and test fails) in grumpy smarty mode.

Hence this re-fixes in by moving the check to only be done for the from
(the sms Upload form) that assigns it

eileenmcnaughton committed:
Remove last pay_later_text isset
